tests/v3/test_metadata/test_v3.py

```diff
@@ -165,6 +167,31 @@ def test_parse_fill_value_invalid_type_sequence(fill_value: Any, dtype_str: str)
         parse_fill_value(fill_value, dtype)
 
 
+def test_parse_fill_value_bytes():
+    result = parse_fill_value("", dtype=np.dtype("S6"))
+    assert result == np.bytes_("")
+
+
+@pytest.mark.parametrize("fill_value", [None, np.bytes_(b"")])
+def test_fill_value_bytes(fill_value: Any) -> None:
+    md = ArrayV3Metadata(
+        shape=(4,),
+        data_type=np.dtype("S6"),
```

Comment:
🤦 - we still don't have a string dtype in the spec! In https://github.com/zarr-developers/zarr-python/pull/2209, I implement a dtype validator that would raise an error if this was passed in.